### PR TITLE
feat: set implicit condition version on azurerm_role_assignment

### PR DESCRIPTION
**Pull request by logan-bobo**
I noticed when using `condition` that `condition_version` is required for resource creation. However when checking out the [Microsoft documentation ](https://learn.microsoft.com/en-us/azure/role-based-access-control/conditions-faq) I noticed the following. 

```
conditionVersion: A condition version number. Defaults to 2.0 and is the only publicly supported version.
```

This change it to propose that on resource creation of `azurerm_role_assignment`, if condition is set we implicitly set `condition_version` to be "2.0" as this is the default and only supported version. 

Happy to discuss and test, feedback would be appreciated.

### CODE REVIEW

**Review by katbyte — CHANGES_REQUESTED** [review]
test failure
```
------- Stdout: -------
=== RUN   TestAccRoleAssignment_implicitCondition
=== PAUSE TestAccRoleAssignment_implicitCondition
=== CONT  TestAccRoleAssignment_implicitCondition
    testcase.go:113: Step 1/2 error: Error running pre-apply refresh: exit status 1
        
        Error: Missing required argument
        
          with azurerm_role_assignment.test,
          on terraform_plugin_test.tf line 37, in resource "azurerm_role_assignment" "test":
          37:   condition            = "@Resource[Microsoft.Storage/storageAccounts/blobServices/containers:ContainerName] StringEqualsIgnoreCase 'foo_storage_container'"
        
        "condition": all of `condition,condition_version` must be specified
--- FAIL: TestAccRoleAssignment_implicitCondition (3.10s)
FAIL
```

File: internal/services/authorization/role_assignment_resource.go [hunk 1]
```diff
@@ -136,17 +136,14 @@ func resourceArmRoleAssignment() *pluginsdk.Resource {
...
 				ValidateFunc: validation.StringIsNotEmpty,
 			},
 
 			"condition_version": {
 				Type:         pluginsdk.TypeString,
 				Optional:     true,
 				ForceNew:     true,
-				RequiredWith: []string{"condition"},
 				ValidateFunc: validation.StringInSlice([]string{
-					"1.0",
```
Comment (tombuildsstuff) [inline internal/services/authorization/role_assignment_resource.go:149]: Whilst `2.0` is defaulted/supported for new resources (although it appears that's been removed from the referenced document at this point) any existing Role Assignments provisioned using `1.0` that's managed by Terraform will fail if we remove this - as such can we add `1.0` back in here?

Comment (logan-bobo) [reply internal/services/authorization/role_assignment_resource.go:149]: Yeah I will add this back in now

**Comment by manicminer** [comment]
Getting 2 failing tests with the same error:

* TestAccRoleAssignment_condition
* TestAccRoleAssignment_implicitCondition

Screenshot 2024-04-17 at 13 29 17

**Comment by stephybun** [comment]
Closing since we've not heard back. @logan-bobo when you get a chance to pick this back up again let us know and we can re-open this!

**Pull request closed without merging**